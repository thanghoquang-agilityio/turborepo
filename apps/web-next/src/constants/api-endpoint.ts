export const API_ENDPOINT = {
  PRODUCTS: 'products',
  PRODUCT_VARIANTS: 'product-variants',
  CATEGORIES: 'categories',
  CARTS: 'carts',
  USERS: 'users',
}

export const USER_ID_DEFAULT = "1"

export const API_URL =
  process.env.NEXT_PUBLIC_API_URL || 'https://api.example.com'
export const API_IMAGE_URL =
  process.env.NEXT_PUBLIC_API_IMAGE_URL || 'https://images.example.com'
