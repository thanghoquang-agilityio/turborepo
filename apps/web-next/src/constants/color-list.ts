export const COLOR_ITEMS = [
  {
    color: 'wheat',
    backgroundColor: 'bg-warning-300',
    borderColor: 'border-warning-300',
  },
  {
    color: 'yellow green',
    backgroundColor: 'bg-success-200',
    borderColor: 'border-success-200',
  },
  {
    color: 'slate blue',
    backgroundColor: 'bg-secondary-400',
    borderColor: 'border-secondary-400',
  },
  {
    color: 'violet',
    backgroundColor: 'bg-danger-200',
    borderColor: 'border-danger-200',
  },
  {
    color: 'salmon',
    backgroundColor: 'bg-danger-300',
    borderColor: 'border-danger-300',
  },
  {
    color: 'dark green',
    backgroundColor: 'bg-success-300',
    borderColor: 'border-success-300',
  },
]

export const COLOR_ITEM_DEFAULT = { backgroundColor: '', borderColor: '' }
