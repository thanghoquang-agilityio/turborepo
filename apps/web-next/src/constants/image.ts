export const IMAGE_NOT_AVAILABLE = 'UnavailableImage'
export const SRC_IMAGE_NOT_AVAILABLE = '/images/image-not-available.webp'

export const SRC_BANNER_HEADPHONE = '/images/banner-headphone.webp'

export const IMAGE_DEFAULT = {
  name: '',
  url: SRC_IMAGE_NOT_AVAILABLE,
}
