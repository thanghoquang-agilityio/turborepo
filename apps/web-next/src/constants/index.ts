export * from './api-endpoint'
export * from './color-list'
export * from './image'
export * from './message'
export * from './navbar'
export * from './pagination'
export * from './product'
export * from './regex'
export * from './relationship'
export * from './router'
export * from './styles'
