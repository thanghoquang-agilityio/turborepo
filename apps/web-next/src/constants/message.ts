export const CART_MESSAGES = {
  ADD: {
    SUCCESS: 'You have successfully added to your Cart.',
    FAILED: 'An error occurred while adding to your Cart. Please try again.',
    WARNING: 'Your cart already contains all products in stock.',
  },
  UPDATE: {
    SUCCESS: 'You have successfully updated to your Cart.',
    FAILED: 'An error occurred while updating to your Cart. Please try again.',
  },
  REMOVE: {
    SUCCESS: 'You have successfully removed from your Cart.',
    FAILED:
      'An error occurred while removing from your Cart. Please try again.',
  },
}
