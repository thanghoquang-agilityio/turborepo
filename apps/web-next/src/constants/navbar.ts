export const NAVBAR_LIST = [
  {
    id: '1',
    name: 'Category',
    value: 'category',
  },
  {
    id: '2',
    name: 'Brand',
    value: 'brand',
  },
  {
    id: '3',
    name: 'Contact',
    value: 'contact',
  },
  {
    id: '4',
    name: 'FAQâ€™s',
    value: 'frequently-asked-question',
  },
]
