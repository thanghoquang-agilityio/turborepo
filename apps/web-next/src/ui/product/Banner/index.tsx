import { memo } from 'react'

import { Button, CustomImage, Text } from '@/components'

interface BannerProps {
  description: string
  image: string
  onBuyNow?: () => void
}

const Banner = memo(({ description, image, onBuyNow }: BannerProps) => {
  const name = description.split('')
  const altImage = name[name.length - 1]

  return (
    <div className='relative flex flex-col items-center gap-4 rounded-none bg-linear-purple pt-[20px] dark:bg-linear-black-green sm:h-[180px] sm:items-start sm:gap-[22px] sm:pl-[75px] md:h-[287px] md:gap-[44px] md:pt-[50px] lg:pl-[135px] lg:pt-[65px] 2xl:rounded-large'>
      <Text className='max-w-[261px] whitespace-pre-wrap text-2xl font-bold text-primary-300 md:max-w-[377px] md:text-5xl md:leading-10'>
        {description}
      </Text>
      <div className='flex items-center'>
        <Button
          variant='solid'
          size='md'
          additionalClass='max-w-[136px]'
          onClick={onBuyNow}
          aria-label='Buy Now'
        >
          Buy Now
        </Button>
        <CustomImage
          src={image}
          alt={altImage}
          className='top-[18px] h-[104px] w-[96px] sm:absolute sm:right-[68px] sm:h-[161px] sm:w-[146px] md:h-[269px] md:w-[244px] lg:right-[178px]'
          width={244}
          height={269}
        />
      </div>
    </div>
  )
})

Banner.displayName = 'Banner'
export default Banner
