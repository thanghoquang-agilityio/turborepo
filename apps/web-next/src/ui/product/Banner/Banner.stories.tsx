import type { Meta, StoryObj } from '@storybook/react'

import { SRC_BANNER_HEADPHONE } from '@/constants'

import Banner from '../Banner'

const meta = {
  title: 'Components/Banner',
  tags: ['autodocs'],
  component: Banner,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof Banner>

export default meta
type Story = StoryObj<typeof Banner>

export const Default: Story = {
  decorators: [
    (Story) => (
      <div className='w-[1024px] min-w-[375px]'>
        <Story />
      </div>
    ),
  ],

  args: {
    image: SRC_BANNER_HEADPHONE,
    description: 'Grab Upto 50% Off On Selected Headphone',
  },
}
