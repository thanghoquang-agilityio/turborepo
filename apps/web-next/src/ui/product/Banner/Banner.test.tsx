import { render } from '@testing-library/react'

import { SRC_BANNER_HEADPHONE } from '@/constants'

import Banner from '.'

describe('Banner', () => {
  const renderComponent = () =>
    render(
      <Banner
        image={SRC_BANNER_HEADPHONE}
        description='Grab Upto 50% Off On Selected Headphone'
      />
    )

  it('should match snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
