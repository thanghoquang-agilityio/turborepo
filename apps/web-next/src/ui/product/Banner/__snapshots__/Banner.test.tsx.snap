// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match snapshot 1`] = `
<div>
  <div
    class="relative flex flex-col sm:items-start items-center bg-linear-purple dark:bg-linear-black-green 2xl:rounded-large rounded-none md:h-[287px] sm:h-[180px] md:gap-[44px] sm:gap-[22px] gap-4 lg:pt-[65px] md:pt-[50px] pt-[20px] lg:pl-[135px] sm:pl-[75px]"
  >
    <p
      class="  no-underline text-ellipsis whitespace-nowrap overflow-hidden text-primary-300 font-bold md:text-5xl text-2xl md:max-w-[377px] max-w-[261px] md:leading-10 whitespace-pre-wrap"
    >
      Grab Upto 50% Off On Selected Headphone
    </p>
    <div
      class="flex items-center"
    >
      <button
        aria-label="Buy Now"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-primary-300 text-background-100 sm:py-4 sm:min-w-[136px] sm:h-12 sm:text-base sm:rounded-[50px] py-2 min-w-20 h-9 text-sm rounded-[30px] max-w-[136px]"
        type="button"
      >
        Buy Now
      </button>
      <img
        alt="e"
        class="sm:absolute top-[18px] lg:right-[178px] sm:right-[68px] md:w-[244px] md:h-[269px] sm:w-[146px] sm:h-[161px] w-[96px] h-[104px]"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="269"
        src="http://localhost/_next/image?url=%2Fimages%2Fbanner-headphone.webp&w=640&q=75"
        srcset="/_next/image?url=%2Fimages%2Fbanner-headphone.webp&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fbanner-headphone.webp&w=640&q=75 2x"
        style="color: transparent; object-fit: cover;"
        width="244"
      />
    </div>
  </div>
</div>
`;
